### PR TITLE
1D Seasonal updates

### PR DESCRIPTION
**Pull request by aburrell**
Added 1-dimensional median calculation to seasonal averaging routine

### CODE REVIEW

**Comment by aburrell** [comment]
Ug, I should add some tests before this is merged.  in the travis fail, it's all the normal things.

**Comment by aburrell** [comment]
The ssnl branch had no unit tests, adding full coverage is out of scope for this pull request.  Once the new routines are fully tested this should be merged.

**Comment by rstoneback** [comment]
There are tests for seasonal under test_avg.py. That file retains an old name from a previous organization.

**Comment by rstoneback** [comment]
My ICON work is taking a little longer than I originally anticipated. Once I get the TIEGCM coupling working in the simulator like it does outside, I'm clear for some pysat review work.

**Comment by jklenzing** [comment]
Asking ahead for the #124 branch:  There's a 1D median function in utils (soon to be utils.stats) that seems to be fairly similar to this function.  I think that the updated routine here is a better home for this feature.  Should I go ahead and remove it from utils?

**Comment by aburrell** [comment]
I'm not sure.  My initial thought is to put the median routines here in utils.  However, until we decide on the fate of ssnl, I agree that it is a better place for things.

**Comment by jklenzing** [comment]
Leaving it in place for now, but not writing a unit test for it as part of #164.

File: pysat/ssnl/avg.py [hunk 1]
```diff
@@ -279,3 +464,106 @@ def _core_mean(inst, data_label, by_orbit=False, by_day=False, by_file=False):
...
+
+    Parameters
+    ----------
+
+    Returns
+    ------
+
+    Notes
+    -----
+    This is an overcomplicated way of doing this.  Try and simplify later
```
[the remark this answers is not in the document]
Comment (rstoneback) [reply pysat/ssnl/avg.py]: The seasonal median was the motivating application for pysat. Here lies the oldest code :)

File: pysat/ssnl/avg.py [hunk 2]
```diff
@@ -11,6 +11,191 @@
...
+                            # take the data (already filtered by x), select the
+                            # data, put it in a list, and extend the deque
+                            idata = inst.data.iloc[xindex]
+                            ans[zk][xi].extend(idata[data_label[zk]].tolist())
+
+    # Calculate the 1D median
+    return _calc_1d_median(ans, data_label, binx, xarr, zarr, numx, numz,
+                           returnData)
+
+def _calc_2d_median(ans, data_label, binx, biny, xarr, yarr, zarr, numx, numy,
```
[the remark this answers is not in the document]
Comment (aburrell) [reply pysat/ssnl/avg.py]: Do you mean the 2D function?

File: pysat/ssnl/avg.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aburrell) [reply pysat/ssnl/avg.py]: I think the best way to do this is to write an ND median/ave/other stat function.  The statistical function could be provided as an input, allowing people to do clever things like find the moments from a Gaussian fit the the data in the desired bin.  Then all the stat functions could go into utils.stats, and the seasonal binning part would live in ssnl.  And if we do ND, then we won't have to keep re-writting things when we eventually want 3D or 4D binning.

File: pysat/tests/test_ssnl_ave.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import datetime as dt
```
[the remark this answers is not in the document]
Comment (aburrell) [reply pysat/tests/test_ssnl_ave.py]: I'm merging them together now.

File: pysat/ssnl/avg.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rstoneback) [reply pysat/ssnl/avg.py]: Sounds like a very nice enhancement.

[the remark this answers is not in the document]
Comment (aburrell) [reply pysat/ssnl/avg.py]: Created an issue for this (#165 )

File: pysat/ssnl/avg.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (aburrell) [reply pysat/ssnl/avg.py]: I think I moved them around when I created things.  There was a merge conflict in this file and now only one exists.

**Comment by aburrell** [comment]
@rstoneback or @jklenzing  ready for approval

**Pull request merged**